### PR TITLE
add env var for firefox, fallback to known locations or which

### LINKED ISSUE #308: All Robot tests started failing on Linux on CI
With the message:

```
Parent suite setup failed:
SessionNotCreatedException: Message: Unable to find a matching set of capabilities
```

Which seems to be connected to selenium/firefox/gecko, see:
- https://stackoverflow.com/questions/43713445/selenium-unable-to-find-a-matching-set-of-capabilities-despite-driver-being-in
= https://stackoverflow.com/questions/47782650/selenium-common-exceptions-sessionnotcreatedexception-message-unable-to-find-a/47785513

### Linux:

> firefox                   79.0                 he1b5a44_1    conda-forge
> selenium                  3.141.0         py36h8c4c3a4_1001
> robotframework-seleniumlibrary 4.5.0              pyh9f0ad1d_0    conda-forge

### MacOS

> firefox                   78.0esr              h4a8c4bd_0    conda-forge
> selenium                  3.141.0         py36h37b9a7d_1001    conda-forge
> robotframework-seleniumlibrary 4.5.0              pyh9f0ad1d_0    conda-forge

### Windows

> firefox                   79.0                 ha925a31_1    conda-forge
> selenium 3.141.0 py36h8c4c3a4_1001
> robotframework-seleniumlibrary 4.5.0 pyh9f0ad1d_0 conda-forge

@bollwyvl any ideas?

### PR DESCRIPTION
**Pull request by bollwyvl**
## References

- fixes #308

## Code changes

- [x] adds a `$FIREFOX_BINARY` environment variable (with some fallbacks) to use when creating webdriver
  - newest `geckodriver` performs an (actually quite helpful, i guess) check whether `firefox_binary` is _in fact_ a firefox binary
  - our test environment in CI uses firefoxen delivered by conda-forge, which use wrapper scripts
- [x] add note to CONTRIBUTING

## User-facing changes

- n/a

## Backwards-incompatible changes

- n/a

## Chores

- [x] linted
- [x] tested
- [x] documented
- [ ] changelog entry

### CODE REVIEW

**Comment by bollwyvl** [comment]
Some of the tests have started, good sign!

**Comment by bollwyvl** [comment]
Everything but windows (and 10 minutes of them, already) are passing...

File: py_src/jupyter_lsp/tests/conftest.py [hunk 1]
```diff
@@ -3,13 +3,14 @@
 import shutil
 from typing import Text
 
-# local imports
```
Comment (krassowski) [inline py_src/jupyter_lsp/tests/conftest.py:6]: Was it manual or do we have different versions of isort/lint/config?

**Comment by krassowski** [comment]
Thank you very much!

**Comment by bollwyvl** [comment]
I did a full clean and install of the binder env+test stuff, no doubt
leading to the lint changes.

We can start locking stuff down harder... Or just be more intentional about
it.

I've been meaning to experiment with conda-lock such that we could:
- do the full conda solve off-line
- check them in (list of urls)
- not even involve the solver in CI
  - save some time
  - be able to cache the downloaded files

We'd probably _still_ want one of the excursions to be unpinned, as i'd
much rather have ci tell us stuff is going to be broken for users than lots
of green check marks... But this was a doozy.

**Pull request merged**